x-pack/plugins/ingest_manager/server/services/agents/status.ts

```diff
@@ -56,9 +56,8 @@ export async function getAgentStatusForConfig(
 async function getEventsCount(soClient: SavedObjectsClientContract, configId?: string) {
   const { total } = await soClient.find({
     type: AGENT_EVENT_SAVED_OBJECT_TYPE,
-    filter: configId
-      ? `${AGENT_EVENT_SAVED_OBJECT_TYPE}.attributes.config_id:"${configId}"`
-      : undefined,
+    searchFields: ['config_id'],
+    search: configId,
```

Comment:
Does this work even when `configId` is `undefined`?

Reply:
I believe so. I added some logging to and it gave
```
getEventsCount { config_id: undefined,
  searchFields: [ 'config_id' ],
  search: undefined }
getEvents total { total: 0 }
```

```
getEventsCount { config_id: undefined,
  searchFields: [ 'config_id' ],
  search: undefined }
getEvents total { total: 8 }
```